src/tcp/Compression.jl

```diff
@@ -0,0 +1,54 @@
+using CodecLz4
+
+@enum Compression::UInt8 begin
+    COMPRESSION_NONE = 0
+    COMPRESSION_CHECKSUM_ONLY = 0x02
+    COMPRESSION_LZ4 = 0x82
+end
+
+Compression(flag::Bool)::Compression = flag ? COMPRESSION_LZ4 : COMPRESSION_NONE
+
+function Compression(name::String)::Compression
+    if lowercase(name) == "lz4"
+        return COMPRESSION_LZ4
+    elseif lowercase(name) == "checksum_only"
+        return COMPRESSION_CHECKSUM_ONLY
+    end
+    error("unkown compression mode: $(name)")
+end
+
+"""compress data according to the compression mode"""
+function compress(mode::Compression, data)::Vector{UInt8}
+    return if mode == COMPRESSION_NONE || mode ==COMPRESSION_CHECKSUM_ONLY
+        data
+    elseif mode == COMPRESSION_LZ4
+        lz4_compress(data)
```

Comment:
Let's also `GC.@preserve` here. This is a high-level function that takes a Julia type and this arguably isn't our responsibility in principle (should be done by the LZ4 library), but the library [fails to do this correctly](https://github.com/JuliaIO/CodecLz4.jl/blob/092191810862b4b94fe44a939e39e1095013013f/src/simple_compression.jl#L6). It's also possible that my understanding of FFI and GC interaction is incorrect, but I'd rather not risk having to debug a hard to reproduce memory corruption bug caused by the GC freeing our buffer halfway through later on.